### DISCUSSION TITLE
Gitlab + Cypress + MSW - msw doesn't seem to be intercepting requests

### DISCUSSION THREAD

**Discussion by mlaw93** [mswjs/msw#1055] [Q&A]
I have a Vue app that uses Cypress + MSW, and locally it runs perfect. I have attempted to deploy it in our gitlab pipeline but during the test phase it does not seem that msw is intercepting the calls. In the gitlab pipeline it is running in a docker container and it does attempt to run the test but the data is not populating as a result of the msw intercept the way it does locally.

Any advice? Someone have a sample project that uses Cypress with MSW in a gitlab pipeline?

**Comment by mikekubn** [comment]
- Have you verified that `msw` has started in the pipelines on GitLab? I had similar issue, when the tests passed successfully in local environment. But when I pushed the code to GitLab, the tests ended up failing. 

- The problem was in loading `mockServiceWorker.js` where on localhost it was loaded from `/public` but on GitLab tests were running on production build and `msw` was missing in `/dist`.

**Comment by kettanaito** [comment]
Hey, @mlaw93. 

Could you provide some additional info about your issue?

1. What kind of tests are we talking about? Which environment do they run in (browser/Node.js)?
2. Could you show us your request handlers? People often forget about environmental variables they use in their request URLs, which breaks CI, where those are not set. 
3. Can you inspect the state of that CI machine? Are there any runtime errors during the tests? It may happen that due to the Docker, the location of the worker script or the address of your application differ. 

The issue is highly likely to do with the CI setup you have. There's an element or two which alter your application behavior and result in the mocking not working as expected.
